spec/book_spec.rb

```diff
@@ -20,4 +20,33 @@
     book = Book.new('My Title', 'My Author')
     expect(book.author).to eq('My Author')
   end
+
+  # Test for add_rental
+describe '#add_rental' do
+  it 'should add the rental to rental array' do
+    book = Book.new('Hello world', 'Joe Ray')
+    person = Person.new(name: 'Hello world', age: 37, parent_permission: true)
+    rental_date = '2022-11-11'
+
+    expect do
+      book.add_rental(rental_date, person)
+    end.to change { book.rentals.length }.by(1)
+
+    expect(book.rentals.last[:date]).to eq(rental_date)
+    expect(book.rentals.last[:book]).to eq(book.to_h)
+    expect(book.rentals.last[:person]).to eq(person.to_h)
+  end
+end
+
```

Comment:
[REQUIRED] Hello team, great job so far but as suggested by the previous review it seems some methods have not been tested.
Let's test the `add_student` method for this class as stated in the project requirements.